src/criteria.jl

```diff
@@ -297,3 +297,30 @@ update(criterion::Patience, loss) = (loss=loss, n_increases=0)
 end
 
 done(criterion::Patience, state) = state.n_increases == criterion.n
+
+"""
+    MaximumChecks(n::Int)
+
+$STOPPING_DOC
+
+A stop is triggered by `n` consecutive checking of done.
+
+"""
+mutable struct MaximumChecks <: StoppingCriterion
+    n::Int
+    function MaximumChecks(n::Int)
+        n > 0 ||
+            throw(ArgumentError("The maxchecking level `n` must be positive. "))
+        return new(n)
+    end
+end
+MaximumChecks(; n) = MaximumChecks(n)
+
+update(criterion::MaximumChecks, loss) = (loss=loss, n_increases=0)
```

Comment:
I found `n_increases` a confusing name for this variable. Maybe `n_checks` is better? 

I suggest you also drop `loss` from the state, as it's not really needed (compare `TimeLimit`).

Reply:
Yes, you are right. First, in a first [commit](https://github.com/ablaom/EarlyStopping.jl/pull/3/commits/450c239759be4f3a4dca09bfcd23567913a4dc6d) I updated the name. Later, I have update the state to be an Int in [following commit](https://github.com/ablaom/EarlyStopping.jl/pull/3/commits/b4bc29066901548ea3319f1909ff270bab9b6914).